dashboards-observability/.cypress/integration/1_event_analytics.spec.js

```diff
@@ -787,6 +863,249 @@ describe('Renders chart and verify Toast message if X-axis and Y-axis values are
+    cy.get('p.euiComboBoxPlaceholder').click({ force: true });
+    cy.get('.euiComboBoxOption__content').eq(0).click();
+    cy.get('.euiResizablePanel.euiResizablePanel--collapsible.euiResizablePanel--middle .euiButton__text')
+    .contains('Update chart')
+    .click();
+    cy.get('.euiTextColor.euiTextColor--subdued').contains('No results found').should('exist');
+  });  
+});
+
+describe('Render Pie chart for Legend and single color contrast change', () => {
+  beforeEach(() => {
+    landOnEventVisualizations();
+  });
+  it('Render Pie chart and verify legends for Position Right and Bottom', () => {
+    renderPieChart();
+    cy.get('[data-text="Right"]').should('have.text', 'Right');
+    cy.get('[data-text="Right"] [data-test-subj="v"]').should('have.attr', 'checked');
+    cy.get('[data-text="Bottom"]').should('have.text', 'Bottom').click();
+    cy.get('[data-text="Bottom"] [data-test-subj="h"]').should('not.have.attr', 'checked');
+    cy.get('[data-test-subj="visualizeEditorRenderButton"]').click({ force: true,multiple: true });
+  });
+
+  it('Render Pie chart and verify legends for Show and Hidden', () => {
+    renderPieChart();
+    cy.get('[data-text="Show"]').should('have.text', 'Show');
+    cy.get('[data-text="Show"] [data-test-subj="show"]').should('have.attr', 'checked');
+    cy.get('[data-text="Hidden"]').should('have.text', 'Hidden').click();
+    cy.get('[data-text="Hidden"] [data-test-subj="hidden"]').should('not.have.attr', 'checked');
+    cy.get('[data-test-subj="visualizeEditorRenderButton"]').click({ force: true,multiple: true });
+  });
+
+  it('Renders Pie chart with single color', () => {
+    renderPieChart();
+    cy.get('.euiIEFlexWrapFix').eq(3).contains('Chart Styles').should('exist');
+    cy.get('[data-test-subj="comboBoxInput"]').eq(3).click();
+    cy.get('[name="Pie"]').click();
+    cy.get('.euiSuperSelectControl').click();
+    cy.get('.euiContextMenuItem.euiSuperSelect__item.euiSuperSelect__item--hasDividers').eq(1).click();
+    cy.get('.euiFlexItem.euiFlexItem--flexGrowZero .euiButton__text').eq(2).click();
+    cy.wait(delay);
+  });
+});
+
+describe('Renders heatmap chart for Chart Style', () => {
+  beforeEach(() => {
+    landOnEventVisualizations();
+    querySearch(TEST_QUERIES[4].query, TEST_QUERIES[4].dateRangeDOM);
+    cy.get('[data-test-subj="configPane__vizTypeSelector"] [data-test-subj="comboBoxInput"]').click();
+    cy.get('[data-test-subj="comboBoxOptionsList "] button span').contains('Heatmap').click();
+  });
+
+  it('Renders heatmap chart with default Color Mode and Scheme', () => {
+    cy.get('.ewdrag.drag.cursor-ew-resize').should('be.visible');
+    cy.get('g.g-gtitle text[data-unformatted|="avg(bytes)"]').should('exist');
+  });
+
+  it('Renders heatmap chart with default Chart Style and Z-axis count()', () => {
+    cy.get('#configPanel__value_options [data-test-subj="comboBoxInput"]').click();
```

Comment:
this one and below, test case can be combined and only z-axis text can be changed.

Reply:
Ok, will combine these 2 test case into one